Reimplement and rename get_annotation function

standage opened the pull request:
This update replaces the `get_annotation` function with the new `collapse_tandem_repeat` and `collapse_all_repeats` functions.

I tried several ways to clean up the regex-based splitting approach, but this approach leads to unintuitive off-by-one errors for internal tandem repeats (i.e. 10 empty strings representing 11 copies of the repeat). Correcting for this resulted in opaque code any way I sliced it.

So I started from scratch and went with a recursive function that finds the first instance of the repeat in the sequence, collapses it, and then calls itself to repeat the process on the remaining sequence. This approach is a bit more concise and clear, and (most importantly) it works perfectly as a drop-in replacement for `get_annotation`!

I also added some doctests to the docstrings of the two new functions, so I updated `make test` in the Makefile to find and run these doctests.

standage commented:
Ready for review and merge!

standage commented:
Also, renamed the `loci_need_split_anno` function to `collapse_repeats_by_length`. Made some minor changes to variable names, but otherwise left that function alone.

rnmitchell commented:
Ok, are you done changing things? I didn't want to merge it if you were still working on it.

All the tests passed and I used the ~1300 sequences we have "known" bracketed annotation for and all matched as well.

standage commented:
> Ok, are you done changing things? I didn't want to merge it if you were still working on it.

Well, I'm on a roll. 😀 Since you're actively responding, let me get one more change in there. Then it will be mergeable.

rnmitchell commented:
hahaha exactly why I asked!

standage commented:
Ok, this last commit renamed the `split_string` method to `sequence_to_bracketed_form`, updated all the places in the code where this was called, and changed a few long variable names to get everything to fit on one 99-character line.

rnmitchell commented:
NOW are you ready? ;)

standage commented:
![doit](https://user-images.githubusercontent.com/566823/83063059-9317bb80-a02d-11ea-9143-55441a8d1036.gif)
